jupyterhub/apihandlers/users.py

```diff
@@ -150,6 +150,12 @@ def patch(self, name):
             raise web.HTTPError(404)
         data = self.get_json_body()
         self._check_user_model(data)
+        # check if the new name is already taken
+        new_name = self.find_user(data['name']) if 'name' in data else None
+        if new_name is not None:
```

Comment:
The presence of the `name` field in the request indicates that a name change is requested. It doesn't make sense to ask the user to change its name to the current value.